featurebyte/query_graph/transform/entity_extractor.py

```diff
@@ -0,0 +1,78 @@
+"""
+This module contains graph entity extraction related classes.
+"""
+from typing import Any, List, Set, Tuple
+
+from dataclasses import dataclass, field
+
+from bson import ObjectId
+
+from featurebyte.query_graph.node import Node
+from featurebyte.query_graph.node.generic import LookupNode
+from featurebyte.query_graph.node.mixin import BaseGroupbyParameters
+from featurebyte.query_graph.transform.base import BaseGraphExtractor
+
+
+class EntityExtractorBranchState:
+    """EntityExtractorBranchState class"""
+
+
+@dataclass
+class EntityExtractorGlobalState:
+    """EntityExtractorGlobalState class"""
+
+    entity_ids: Set[ObjectId] = field(default_factory=set)
+
+
+class EntityExtractor(
+    BaseGraphExtractor[
+        EntityExtractorGlobalState, EntityExtractorBranchState, EntityExtractorGlobalState
+    ]
+):
+    """EntityExtractor class"""
+
+    def _pre_compute(
+        self,
+        branch_state: EntityExtractorBranchState,
+        global_state: EntityExtractorGlobalState,
+        node: Node,
+        input_node_names: List[str],
+    ) -> Tuple[List[str], bool]:
+        skip_input_nodes = False
+        if isinstance(node.parameters, BaseGroupbyParameters):
+            # if groupby node has entity_ids, skip further exploration on input nodes
+            if node.parameters.entity_ids:
```

Comment:
It is also possible to use an empty list as the groupby keys to create purely time based features without any entity. Will that be handled correctly by this condition?

Reply:
If it is possible to create such a feature, we should move this `skip_input_nodes` outside. Will update the logic here.

Reply:
Updated the logic here & added test to cover this case.